src/server/index.ts

```diff
@@ -64,7 +112,46 @@ function startMatch(players: [User, User]) {
 	movePlayerToBattle(players[1], players[0]);
 }
 
-export function initServer(server: Server) {
+
+function determineWinner(players: User[], action1: Action, action2: Action)
+{
+	if (action1 === action2) return 'tie';
+
+	else 
+	{
+		if (action1 === 'rock')
+		{
+			if (action2 === 'scissors') return players[0];
+
+			return players[1];
+		}
+
+		else if (action1 === 'paper')
+		{
+			if (action2 === 'rock') return players[0];
+
+			return players[1];
+		}
+
+		else if (action1 === 'scissors')
+		{
+			if (action2 === 'paper') return players[0];
+
+			return players[1];
+		}
+
+
+		else 
+			console.log('something went wrong in determineWinner(), defaulting to a tie');
```

Comment:
Even if it's just one line, it's recommended to always have curly brackets for readability :)